vectordraw.js

```diff
@@ -0,0 +1,282 @@
+'use strict';
+
+var VectorDraw = function(element_id, settings) {
+    var default_settings = {
+        width: 500,
+        height: 350,
+        background: null,
+        bounding_box: [-10, 10, 10, -10],
+        vectors: [],
+        points: []
+    };
+
+    this.board = null;
+    this.dragged_vector = null;
+    this.history_stack = {undo: [], redo: []};
+    this.settings = _.extend(default_settings, settings);
+    this.element = $('#' + element_id);
+
+    this.element.on('click', '.reset', this.reset.bind(this));
+    this.element.on('click', '.add-vector', this.addVectorFromList.bind(this));
+    this.element.on('click', '.undo', this.undo.bind(this));
+    this.element.on('click', '.redo', this.redo.bind(this));
+
+    this.render();
+};
+
+VectorDraw.prototype.template = _.template([
+    '<div class="jxgboard" style="width:<%= width %>px; height:<%= height %>px;" />',
+    '<div class="menu">',
+    '    <div>',
+    '        <select>',
+    '        <% vectors.forEach(function(vec, idx) { %>',
+    '            <option value="<%= idx %>"><%= vec.description %></option>',
+    '        <% }) %>',
+    '        </select>',
+    '        <button class="add-vector">Add Selected Force</button>',
+    '        <button class="reset">Clear All</button>',
+    '        <button class="undo">Undo</button>',
+    '        <button class="redo">Redo</button>',
+    '    </div>',
+    '</div>'
+].join('\n'));
+
+VectorDraw.prototype.render = function() {
+    this.element.html(this.template(this.settings));
+    // Assign the jxgboard element a random unique ID,
+    // because JXG.JSXGraph.initBoard needs it.
+    this.element.find('.jxgboard').prop('id', _.uniqueId('jxgboard'));
+    this.createBoard();
+};
+
+VectorDraw.prototype.createBoard = function() {
+    var id = this.element.find('.jxgboard').prop('id');
+
+    this.board = JXG.JSXGraph.initBoard(id, {
+        boundingbox: this.settings.bounding_box,
+        axis: false,
```

Comment:
@mtyaka nit: there might be use case where axis are needed, so it would be nice to allow switching them on via vectordraw_settings